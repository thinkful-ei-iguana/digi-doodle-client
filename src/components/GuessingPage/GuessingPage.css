@font-face {
    font-family: "cf_crayonsregular";
    src: url("../../Fonts/Crayon/cfcrayons-regular-webfont.woff2") format("woff2"),
      url("../../Fonts/Crayon/cfcrayons-regular-webfont.woff") format("woff");
    font-weight: normal;
    font-style: normal;
  }
.guess-input {
    font-size: 26px;
    margin-top: 7%;
}
.submit-guess {
    border-radius: 10px;
    font-size: 22px;
    color:green;
    border: 0.5px solid #222222;
    margin-left: -1%;
    cursor: pointer;
}
.player-ul {
    display:flex;
    justify-content: space-evenly;
    flex-direction: row;
    margin-left: -3.5%;
    margin-top: 2.2%;
}
li {
    list-style-type: none;
    font-size: 26px;
    margin-top: 1.5%;
}
.player-li1 {
    color: green;
}
.player-li2{
    color: blue;
}
.player-li3{
    color: #f55440;
}
.player-li4{
    color: purple;
}
.disabled-canvas {
    pointer-events: none;
}
.reminder-instructions {
    color: gray;
}
@media screen and (max-width: 500px) {
    li {
        font-size: 16px;
        padding: 0px 5px;
        margin-right: 25px;
    }
    .guess-input {
        font-size: 20px;
    }
    input {
        width: 30%;
    }
    .submit-guess {
        font-size: 18px;
    }
}
@media screen and (max-width: 350px) {
    li {
        font-size: 14px;
    }
    ul {
        margin: 0 8px;
    }
    .guess-input {
        font-size: 20px;
    }
    input {
        width: 30%;
    }
    .submit-guess {
        font-size: 18px;
    }
}
