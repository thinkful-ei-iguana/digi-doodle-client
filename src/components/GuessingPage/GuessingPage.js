import React, { Component } from 'react'
import Canvas from '../../Utils/Canvas/Canvas'
import './GuessingPage.css'
import ColorContext from '../../Context/ColorContext'
import Cookies from 'js-cookie';
import socket from '../../services/socket-service'
import '../../Utils/Canvas/Canvas.css'
import { animateScroll } from "react-scroll";
import PlayAgain from '../PlayAgain/PlayAgain'

export default class GuessingPage extends Component {
    constructor(props) {
        super(props);
        this.state = {
            guess: '',
            username: '',
            players: [],
            score: 0
        }
    }

    static contextType = ColorContext;

    async componentDidMount() {

        let cookie = Cookies.get();
        if (cookie && cookie['digi-doodle-user']) {
            let data = JSON.parse(cookie['digi-doodle-user']);
            await this.setState({
                username: data.username
            });
        }

    }

    handleGuessSubmit = async (ev) => {
        try {
            ev.preventDefault();
            socket.emit('guess', { player: this.state.username, message: this.state.guess });

            // console.log('guess response: ', guess);
            await this.setState({
                guess: ''
            });
        } catch (error) {
            console.error(error)
        }
    }

    handleTextInput = (ev) => {
        this.setState({
            guess: ev.target.value
        })
    }

    scrollToBottom() {
        animateScroll.scrollToBottom({
            containerId: "chatUl"
        });
    }


    //event handler for submit button to validate answer

    render() {
        let disableAttr = this.props.isDrawing ? "canvas-container" : "disabled-canvas"

        let toggleDisplay;

        if (this.context.game && this.context.game.winner) {
            toggleDisplay = <PlayAgain />
        } else {
            toggleDisplay =
                <div>
                    <form className="guess-input" >
                        <label htmlFor="chat-input">Guess goes here: </label>
                        <input type="text" onChange={this.handleTextInput}
                            id="chat-input"
                            value={this.state.guess}
                            required
                            spellCheck="false"
                            maxLength="35"
                        />
                        <button className="submit-guess" type="submit" id="chat-submit" onClick={this.handleGuessSubmit}>&#10004;</button>
                    </form>
                    <div className="chat-window">
                        <ul>

                            {this.context.messages && this.context.messages.map((message, index) => {
                                if (message.player === 'Lobby') {
                                return (
                                    <li className="lobby-message" key={index}>{message.message}</li>
                                    )} else {
                                return (
                                    <li className="player-message" key={index}>{message.player}: {message.message}</li>
                                    )
                                }
                            })}
                        </ul>
                    </div>
                </div>
        }

        return (
            <div>
                <div className={disableAttr}>
                    <Canvas />
                </div>
                <div className="players-container">
                    <ul className="player-ul">
                        {this.context.players && this.context.players.map((player, index) => {
                            return (
                                <li className="player-li" key={index}>
                                    <span>{player.username}</span><br />
                                    <span className="score">{player.score}</span>
                                </li>
                            )
                        })}
                    </ul>
                </div>

                {toggleDisplay}

                <p className="reminder-instructions">15 points to win</p>
            </div>
        )
    }
}