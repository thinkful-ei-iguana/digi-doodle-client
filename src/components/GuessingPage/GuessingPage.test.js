import React from 'react'
import GuessingPage from './GuessingPage'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'


describe(`Guessing component`, () => {

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <GuessingPage />
            , div);
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders without crashing', () => {

        window.HTMLCanvasElement.prototype.getContext = () => { }
        const context = {
            game: {
                id: 'game uuid',
                current_drawer: 'test player',
                current_answer: 'test',
                status: 'waiting for players',
                winner: 'test winner'
            }
        }

        const guessPage = shallow(<GuessingPage />, { context })
        guessPage.setContext({ context })
        expect(guessPage).toMatchSnapshot();

    })

    // it('renders text input with label', () => {
    //     const wrapper = mount(<GuessingPage />)
    //     const label = wrapper.find('label')
    //     const input = wrapper.find('input');
    //     expect(input).toHaveLength(1);
    //     expect(label).toHaveLength(1);
    //     expect(input.prop('type')).toEqual('text')
    // })
})