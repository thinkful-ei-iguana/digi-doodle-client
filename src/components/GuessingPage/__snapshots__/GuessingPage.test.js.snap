// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Guessing component renders without crashing 1`] = `ShallowWrapper {}`;
